### ISSUE TITLE
💅 useSingleVarDeclarator suggests non-working code

### ISSUE THREAD

**Issue by rubiesonthesky** [biomejs/biome#728]
### Environment information

```bash
CLI:
  Version:                      1.3.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.9.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

useSingleVarDeclarator

### Playground link

https://biomejs.dev/playground/?code=IAAgACAAIABjAG8AbgBzAHQACgAgACAAIAAgACAAIABBACAAPQAgADEAMAAwACwACgAgACAAIAAgACAAIABCACAAPQAgADEANQAwACwACgAgACAAIAAgACAAIABEACAAPQAgADEAMgAwACwACgAgACAAIAAgACAAIABFACAAPQAgAEQAIAAvACAAQgAsAAoAIAAgACAAIAAgACAARgAgAD0AIAB7AGgAZQBpAGcAaAB0ADoAIABCACwAIAB3AGkAZAB0AGgAOgAgAEEAfQA7AA%3D%3D

### Expected result

Code should work after lint fix :) 

```diff
-     const
-       A = 100,
-       B = 150,
-       D = 120,
-       E = D / B,
-       F = {height: B, width: A};
+     constA = 100;    constB = 150;    constD = 120;    constE = D / B;    constF = {height: B, width: A};
```

Expected result was 
```ts
    const A = 100;
    const B = 150;
    const D = 120;
    const E = D / B;
    const F = {height: B, width: A};
```

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
